lib/lua/CMakeLists.txt

```diff
@@ -50,13 +53,27 @@ if(LUA_ANSI)
 	set(COMMON_CFLAGS "${COMMON_CFLAGS} -DLUA_ANSI")
 endif(LUA_ANSI)
 
+if (NOT LUA_COMPILE_AS_C)
+	# Force compilation as C++.
+	if(WIN32)
+		set(COMMON_CFLAGS "${COMMON_CFLAGS} -TP")
+	else()
+		set(COMMON_CFLAGS "${COMMON_CFLAGS} -xc++")
+	endif()
+endif(NOT LUA_COMPILE_AS_C)
+
 # COMMON_CFLAGS has no effect without this line
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COMMON_CFLAGS}")
 
 
 # Standard flags to use for each build type.
 if(CMAKE_COMPILER_IS_GNUCC)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -Wall -Wextra -Wshadow -W -pedantic -std=gnu99")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -Wall -Wextra -Wshadow -W -pedantic")
+	if(LUA_COMPILE_AS_C)
+		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
+	else()
+		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu++11")
+	endif()
```

Comment:
I don't know if there's an even more portable way but it should at least do this:
<pre>
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu++${CMAKE_CXX_STANDARD}")
</pre

Reply:
Or maybe you can set a property on the target instead https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD
This may redoing a bit of the build setup if there is not already one.

Reply:
I made the first change you suggested. I wasn't able to get the other stuff working, as I don't know much about CMake. You probably know more, and I think you have edit access, so maybe you can figure something out.

I tried to compile the source files as C++ using the following code, which did not work for me.

```cmake
file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}lib/lua/src/*.c")
set_source_files_properties(${SOURCE_FILES} PROPERTIES LANGUAGE CXX)
```